### PR TITLE
Curl timeout.

### PR DESCRIPTION
**Pull request by sergeypavlenko**
Hi

First of all thanks for the script!

I have a problem on my site with timeout curl. For me once script not complete success, php output info about error "maximum execution time". In script I not find timeout for curl, propose the following solution.

Request CURL may last no more than 2 seconds. Average this time is enough.

### CODE REVIEW

File: src/BotApi.php [hunk 1]
```diff
@@ -194,7 +194,9 @@ public function call($method, array $data = null)
             CURLOPT_URL => $this->getUrl() . '/' . $method,
             CURLOPT_RETURNTRANSFER => true,
             CURLOPT_POST => null,
-            CURLOPT_POSTFIELDS => null
+            CURLOPT_POSTFIELDS => null,
+            CURLOPT_CONNECTTIMEOUT_MS => 2000, // 2 second.
+            CURLOPT_NOSIGNAL => TRUE,
```
Comment (Skaronator) [inline src/BotApi.php]: remove comma on the end

**Comment by sergeypavlenko** [comment]
Why close ?

**Comment by iGusev** [comment]
Сhanging the standard timeout is bad idea. This is not a universal solution.

**Comment by antonygorin** [comment]
I'm not much familiar with your code guys but you could add this param to your module initialization config or add it as a static param to your "transport" class. Or @sergeypavlenko can extend the curl wrapper code if the project architecture allows. 

You should take into account that in case of the sending limit violation telegram hangs the connection for a certain amount of time no matter how long you are waiting for it. That's why it's pretty reasonable to set the lowest value you consider to be enough to deliver all legit messages. 

Tuning the timeout value can make your bot more responsive even in case of a low load. When you are big enough to send millions or dozens of millions messages a day it can be vital.

**Comment by sergeypavlenko** [comment]
Hi @antonygorin 

Thank for good idea. @iGusev maybe add new method or parameter in method call() for override or add new CURL settings. It's will be very nice for me.

**Comment by iGusev** [comment]
@sergeypavlenko this method already exists:
```php
protected function executeCurl(array $options)
{
    curl_setopt_array($this->curl, $options);
    $result = curl_exec($this->curl);
    self::curlValidate($this->curl);
    return $result;
}
```
With your changes:
```php
protected function executeCurl(array $options)
{
    curl_setopt_array($this->curl, array_merge($options, [CURLOPT_CONNECTTIMEOUT_MS => 2000, CURLOPT_NOSIGNAL => TRUE]));
    $result = curl_exec($this->curl);
    self::curlValidate($this->curl);
    return $result;
}

**Pull request closed without merging**